mlflow/transformers/peft.py

```diff
@@ -0,0 +1,58 @@
+"""
+PEFT (Parameter-Efficient Fine-Tuning) is a library for efficiently adapting large pretrained
+models without fine-tuning all of model parameters but only a small number of (extra) parameters.
+Users can define a PEFT model that wraps a Transformer model to apply a thin adapter layer on
+top of the base model. The PEFT model provides almost the same APIs as the original model such
+as from_pretrained(), save_pretrained().
+"""
+_PEFT_ADAPTOR_DIR_NAME = "peft"
+
+
+def is_peft_model(model) -> bool:
+    try:
+        from peft import PeftModel
+    except ImportError:
+        return False
```

Comment:
Shall we raise an exception on missing `peft` package instead of returning a False? I am thinking about the following scenario: User A logged a peft model to mlflow, then user B wants to load the peft model from mlflow. However, user B does not have peft in the dev environment, we should prompt user B to install `peft`.

Reply:
This check is only used for saving/logging, and we can't raise Exception because it blocks logging/saving for non-peft model. For loading scenario, we check it based on the logged model metadata like this, and users will get exception when `get_model_with_peft_adapter` is called but PEFT is not installed.
```
    if peft_adapter_dir := flavor_config.get(FlavorKey.PEFT, None):
        model_and_components[FlavorKey.MODEL] = get_model_with_peft_adapter(
            base_model=model_and_components[FlavorKey.MODEL],
            peft_adapter_path=os.path.join(path, peft_adapter_dir),
        )
 ```

Reply:
gotcha, thanks for the clarification!